pkg/collector/corechecks/oracle-dbm/sysmetrics.go

```diff
@@ -9,59 +9,151 @@ import (
+	"Physical Read Total IO Requests Per Sec":       {DDmetric: "physical_read_total_io_requests", DBM: true},
+	"Physical Reads Direct Lobs Per Sec":            {DDmetric: "physical_reads_direct_lobs", DBM: true},
+	"Physical Read Total Bytes Per Sec":             {DDmetric: "physical_read_total_bytes", DBM: true},
+	"Physical Reads Direct Per Sec":                 {DDmetric: "physical_reads_direct", DBM: true},
+	"Physical Reads Per Sec":                        {DDmetric: "physical_reads"},
+	"Physical Write Bytes Per Sec":                  {DDmetric: "physical_write_bytes", DBM: true},
+	"Physical Write IO Requests Per Sec":            {DDmetric: "physical_write_io_requests", DBM: true},
+	"Physical Write Total Bytes Per Sec":            {DDmetric: "physical_write_total_bytes", DBM: true},
+	"Physical Write Total IO Requests Per Sec":      {DDmetric: "physical_write_total_io_requests", DBM: true},
+	"Physical Writes Direct Lobs Per Sec":           {DDmetric: "physical_writes_direct_lobs", DBM: true},
+	"Physical Writes Direct Per Sec":                {DDmetric: "physical_writes_direct", DBM: true},
+	"Physical Writes Per Sec":                       {DDmetric: "physical_writes"},
+	"Process Limit %":                               {DDmetric: "process_limit", DBM: true},
+	"Redo Allocation Hit Ratio":                     {DDmetric: "redo_allocation_hit_ratio", DBM: true},
+	"Redo Generated Per Sec":                        {DDmetric: "redo_generated", DBM: true},
+	"Redo Writes Per Sec":                           {DDmetric: "redo_writes", DBM: true},
+	"Row Cache Hit Ratio":                           {DDmetric: "row_cache_hit_ratio", DBM: true},
+	"Rows Per Sort":                                 {DDmetric: "rows_per_sort"},
+	"SQL Service Response Time":                     {DDmetric: "service_response_time"},
+	"Session Count":                                 {DDmetric: "session_count"},
+	"Session Limit %":                               {DDmetric: "session_limit_usage"},
+	"Shared Pool Free %":                            {DDmetric: "shared_pool_free"},
+	"Soft Parse Ratio":                              {DDmetric: "soft_parse_ratio", DBM: true},
+	"Temp Space Used":                               {DDmetric: "temp_space_used"},
+	"Total Parse Count Per Sec":                     {DDmetric: "total_parse_count", DBM: true},
+	"Total Sorts Per User Call":                     {DDmetric: "sorts_per_user_call"},
+	"User Commits Per Sec":                          {DDmetric: "user_commits", DBM: true},
+	"User Rollbacks Per Sec":                        {DDmetric: "user_rollbacks"},
+}
+
+func (c *Check) sendMetric(s aggregator.Sender, r SysmetricsRowDB, seen map[string]bool) {
+	if metric, ok := SYSMETRICS_COLS[r.MetricName]; ok {
+		value := r.Value
+		if r.MetricUnit == "CentiSeconds Per Second" {
+			value = value / 100
+		}
+		if !SYSMETRICS_COLS[r.MetricName].DBM || SYSMETRICS_COLS[r.MetricName].DBM && c.dbmEnabled {
```

Comment:
This doesn't have to change now but if DBM is disabled and some metrics are not going to get sent, should we skip the collection of the values entirely?

Reply:
A `WHERE` condition wouldn't be selective. I haven't measured it, but a full table scan might be cheaper than a filter with many `OR` operands. We can do a benchmark at some less busy times.

Reply:
Oh right. This makes total sense to do the filtering in post-processing then.